cachito/workers/pkg_managers/gomod.py

```diff
@@ -156,17 +167,22 @@ def resolve_gomod(app_source_path, request, dep_replacements=None):
 
         bundle_dir = RequestBundleDir(request["id"])
 
-        # Add the gomod cache to the bundle the user will later download
-        tmp_download_cache_dir = os.path.join(temp_dir, RequestBundleDir.go_mod_cache_download_part)
-        if not os.path.exists(tmp_download_cache_dir):
-            os.makedirs(tmp_download_cache_dir, exist_ok=True)
+        if "gomod-vendor" in request.get("flags", []):
+            bundle_dir.gomod_download_dir.mkdir(exist_ok=True, parents=True)
```

Comment:
```suggestion
            # Create an empty gomod cache in the bundle directory so that any Cachito
            # user does not have to guard against this directory not existing
            bundle_dir.gomod_download_dir.mkdir(exist_ok=True, parents=True)
```